cluster-autoscaler/processors/nodegroupset/compare_nodegroups.go

```diff
@@ -56,18 +56,25 @@ type NodeInfoComparator func(n1, n2 *schedulerframework.NodeInfo) bool
 func compareResourceMapsWithTolerance(resources map[apiv1.ResourceName][]resource.Quantity,
 	maxDifferenceRatio float64) bool {
 	for _, qtyList := range resources {
-		if len(qtyList) != 2 {
-			return false
-		}
-		larger := math.Max(float64(qtyList[0].MilliValue()), float64(qtyList[1].MilliValue()))
-		smaller := math.Min(float64(qtyList[0].MilliValue()), float64(qtyList[1].MilliValue()))
-		if larger-smaller > larger*maxDifferenceRatio {
+		if !compareResourceListWithTolerance(qtyList, maxDifferenceRatio) {
 			return false
 		}
 	}
 	return true
 }
 
+func compareResourceListWithTolerance(qtyList []resource.Quantity, maxDifferenceRatio float64) bool {
```

Comment:
It's a minor detail, but for a function that returns a bool I'd prefer a name that makes it obvious what true/false means.

Reply:
I was trying to keep consistent with the `compareResourceMapsWithTolerance`, though I agree, it's not a great name. If I rename one, should I rename the other too? How about `resourceMapsWithinTolerance` and `resourceListWithinTolerance`?

Reply:
Fair point on consistency. Renaming both sgtm.